### PR TITLE
Create generateCarePlan API in FhirOperator

### LINKED ISSUE #656: Implementation of $apply API
**User story:** 
As an implementer I can use the $apply operation to generate a CarePlan for a patient given a PlanDefinition

**Background**
CQL (Clinical Quality Language) is the HL7 standard for providing logic execution for decision support and clinical quality measures. 

Our initial goal with the CQL Library will be to demonstrate over time the ability to:

- Generate a MeasureReport via $measure-evaluate operation 
- Load and execute a custom CQL library (Library resource) for decision logic
- Transform a PlanDefinition to a CarePlan using $apply operation

### PR DESCRIPTION
**Pull request by jingtang10**
**IMPORTANT: All PRs must be linked to an issue (except for extremely trivial and straightforward changes).**

Fixes #656 

**Description**
Create the generateCarePlan API in FhirOperator

_NOTE_ the test case is not functioning pending clarification on the local end points.

**Alternative(s) considered**
NA

**Type**
Feature

**Screenshots (if applicable)**

**Checklist**
- [x] I have read and acknowledged the [Code of conduct](https://github.com/google/android-fhir/blob/master/docs/CODE_OF_CONDUCT.md)
- [x] I have read the [Contributing](https://github.com/google/android-fhir/wiki/Contributing) page
- [x] I have signed the Google [Individual CLA](https://cla.developers.google.com/about/google-individual), or I am covered by my company's [Corporate CLA](https://cla.developers.google.com/about/google-corporate )
- [x] I have discussed my proposed solution with code owners in the linked issue(s) and we have agreed upon the general approach
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the style guide of this project
- [x] I have run `./gradlew check` and `./gradlew connectedCheck` to test my changes locally
- [x] I have built and run the reference app(s) to verify my change fixes the issue and/or does not break the reference app(s)

### CODE REVIEW

File: workflow/src/main/java/com/google/android/fhir/workflow/FhirOperator.kt [hunk 1]
```diff
@@ -67,17 +171,47 @@ class FhirOperator(fhirContext: FhirContext, fhirEngine: FhirEngine) {
...
+      /* userLanguage= */ null,
+      /* userTaskContext= */ null,
+      /* setting= */ null,
+      /* settingContext= */ null,
+      /* mergeNestedCarePlans= */ null,
+      /* parameters= */ Parameters(),
+      /* useServerData= */ null,
+      /* bundle= */ null,
+      /* prefetchData= */ null,
+      dataEndpoint,
```
Comment (JPercival) [inline workflow/src/main/java/com/google/android/fhir/workflow/FhirOperator.kt:212]: The endpoint parameters are only necessary if you're planning to use a "remote" data, content, or terminology source. It just so happens that an external network drive could be considered remote so such a scenario is supported but doesn't seem necessary given that the fhir-engine is effectively an embedded database. Opened a PR to illustrate this.

File: workflow/src/test/java/com/google/android/fhir/workflow/FhirOperatorTest.kt [hunk 2]
```diff
@@ -100,6 +103,27 @@ class FhirOperatorTest {
...
   }
 
+  @Test
+  @Ignore("Refactor the API to accommodate local end points")
+  fun generateCarePlan() = runBlocking {
+    val endpoint =
+      Endpoint()
+        .setAddress("RuleFilters-1.0.0-bundle.json")
+        .setConnectionType(Coding().setCode(Constants.HL7_FHIR_FILES))
+    val dataEndpoint: Endpoint =
```
Comment (JPercival) [inline workflow/src/test/java/com/google/android/fhir/workflow/FhirOperatorTest.kt:113]: See other comments on Endpoints.

File: workflow/src/test/java/com/google/android/fhir/workflow/FhirOperatorTest.kt [hunk 3]
```diff
@@ -100,6 +103,27 @@ class FhirOperatorTest {
...
+      Endpoint()
+        .setAddress("tests-Reportable-bundle.json")
```
Comment (JPercival) [inline workflow/src/test/java/com/google/android/fhir/workflow/FhirOperatorTest.kt:115]: This is a Bundle you'd load "ahead of time" into the fhir engine and access via the FhirDal API

File: workflow/src/main/java/com/google/android/fhir/workflow/FhirOperator.kt (hunk 1 above)
Comment (pld) [reply workflow/src/main/java/com/google/android/fhir/workflow/FhirOperator.kt:212]: Given that, can we make these optional parameters?

Comment (jingtang10) [reply workflow/src/main/java/com/google/android/fhir/workflow/FhirOperator.kt:212]: @pld we should probably just get rid of these altogether in the workflow lib api.

**Comment by jingtang10** [comment]
seeing the following with the latest code change:

```
Unable translate library FHIRHelpers
org.opencds.cqf.cql.engine.exception.CqlException: Unable translate library FHIRHelpers
	at org.opencds.cqf.cql.evaluator.engine.execution.TranslatingLibraryLoader.translate(TranslatingLibraryLoader.java:128)
	at org.opencds.cqf.cql.evaluator.engine.execution.TranslatingLibraryLoader.load(TranslatingLibraryLoader.java:78)
	at org.opencds.cqf.cql.engine.execution.Context.resolveIncludeDef(Context.java:304)
	at org.opencds.cqf.cql.engine.execution.Context.enterLibrary(Context.java:319)
	at org.opencds.cqf.cql.engine.elm.execution.FunctionRefEvaluator.internalEvaluate(FunctionRefEvaluator.java:20)
	at org.opencds.cqf.cql.engine.elm.execution.Executable.evaluate(Executable.java:18)
	at org.opencds.cqf.cql.engine.elm.execution.EqualEvaluator.internalEvaluate(EqualEvaluator.java:75)
	at org.opencds.cqf.cql.engine.elm.execution.Executable.evaluate(Executable.java:18)
	at org.opencds.cqf.cql.engine.elm.execution.QueryEvaluator.evaluateWhere(QueryEvaluator.java:75)
	at org.opencds.cqf.cql.engine.elm.execution.QueryEvaluator.internalEvaluate(QueryEvaluator.java:198)
	at org.opencds.cqf.cql.engine.elm.execution.Executable.evaluate(Executable.java:18)
	at org.opencds.cqf.cql.engine.elm.execution.ExpressionDefEvaluator.internalEvaluate(ExpressionDefEvaluator.java:19)
	at org.opencds.cqf.cql.engine.elm.execution.Executable.evaluate(Executable.java:18)
	at org.opencds.cqf.cql.engine.elm.execution.ExpressionRefEvaluator.internalEvaluate(ExpressionRefEvaluator.java:11)
	at org.opencds.cqf.cql.engine.elm.execution.Executable.evaluate(Executable.java:18)
	at org.opencds.cqf.cql.engine.elm.execution.ExpressionDefEvaluator.internalEvaluate(ExpressionDefEvaluator.java:19)
	at org.opencds.cqf.cql.engine.elm.execution.Executable.evaluate(Executable.java:18)
	at org.opencds.cqf.cql.evaluator.measure.common.MeasureEvaluation.evaluateStratifiers(MeasureEvaluation.java:490)
	at org.opencds.cqf.cql.evaluator.measure.common.MeasureEvaluation.evaluateGroup(MeasureEvaluation.java:458)
	at org.opencds.cqf.cql.evaluator.measure.common.MeasureEvaluation.innerEvaluate(MeasureEvaluation.java:299)
	at org.opencds.cqf.cql.evaluator.measure.common.MeasureEvaluation.evaluate(MeasureEvaluation.java:283)
	at org.opencds.cqf.cql.evaluator.measure.common.MeasureEvaluation.evaluate(MeasureEvaluation.java:107)
	at org.opencds.cqf.cql.evaluator.measure.r4.R4MeasureProcessor.evaluateMeasure(R4MeasureProcessor.java:203)
	at com.google.android.fhir.workflow.FhirOperator.evaluateMeasure(FhirOperator.kt:172)
	at com.google.android.fhir.workflow.FhirOperatorTest$evaluateIndividualSubjectMeasure$1.invokeSuspend(FhirOperatorTest.kt:59)
	at app//kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
	at kotlinx.coroutines.EventLoopImplBase.processNextEvent(EventLoop.common.kt:274)
	at kotlinx.coroutines.BlockingCoroutine.joinBlocking(Builders.kt:85)
	at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking(Builders.kt:59)
	at kotlinx.coroutines.BuildersKt.runBlocking(Unknown Source)
	at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking$default(Builders.kt:38)
	at kotlinx.coroutines.BuildersKt.runBlocking$default(Unknown Source)
	at com.google.android.fhir.workflow.FhirOperatorTest.evaluateIndividualSubjectMeasure(FhirOperatorTest.kt:57)
	at java.base@11.0.11/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base@11.0.11/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base@11.0.11/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base@11.0.11/java.lang.reflect.Method.invoke(Method.java:566)
	at app//org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at app//org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at app//org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at app//org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at app//org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at app//org.robolectric.RobolectricTestRunner$HelperTestRunner$1.evaluate(RobolectricTestRunner.java:575)
	at app//org.robolectric.internal.SandboxTestRunner$2.lambda$evaluate$0(SandboxTestRunner.java:278)
	at app//org.robolectric.internal.bytecode.Sandbox.lambda$runOnMainThread$0(Sandbox.java:89)
	at java.base@11.0.11/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base@11.0.11/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base@11.0.11/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base@11.0.11/java.lang.Thread.run(Thread.java:829)
```

File: workflow/src/test/java/com/google/android/fhir/workflow/FhirOperatorTest.kt [hunk 4]
```diff
@@ -43,29 +42,22 @@ class FhirOperatorTest {
...
-      if (entry.resource.resourceType == ResourceType.Library) {
-        fhirOperator.loadLib(entry.resource as Library)
-      } else {
-        fhirEngine.save(entry.resource)
-      }
-    }
+    loadBundle("/ANCIND01-bundle.json")
+    loadBundle("/RuleFilters-1.0.0-bundle.json")
+    loadBundle("/tests-Reportable-bundle.json")
+    loadBundle("/tests-NotReportable-bundle.json")
```
Comment (jingtang10) [inline workflow/src/test/java/com/google/android/fhir/workflow/FhirOperatorTest.kt:48]: these are causing issues as commented above

**Comment by Tarun-Bhardwaj** [comment]
@f-odhiambo , have these changes been reviewed internally?

**Comment by pld** [comment]
> @f-odhiambo , have these changes been reviewed internally?

We did a review 19 days ago and made comments here and below, https://github.com/google/android-fhir/issues/656#issuecomment-1063598193

**Pull request merged**